iceoryx_posh/source/roudi/memory/iceoryx_roudi_memory_manager.cpp

```diff
@@ -47,7 +47,7 @@ cxx::expected<RouDiMemoryManagerError> IceOryxRouDiMemoryManager::destroyMemory(
 
 PosixShmMemoryProvider* IceOryxRouDiMemoryManager::mgmtMemoryProvider() noexcept
 {
-    return &m_defaultMemory.m_managementShm;
+    return const_cast<PosixShmMemoryProvider*>(const_cast<const IceOryxRouDiMemoryManager*>(this)->mgmtMemoryProvider());
```

Comment:
I guess it is better to return a const pointer, please change the signature to this:
```
const PosixShmMemoryProvider* IceOryxRouDiMemoryManager::mgmtMemoryProvider(() const noexept
```
Then you do not need to const_cast the return value.

Reply:
There is already a const method.

Reply:
Ah right below, nevertheless this function here is dead code in my point of view because nobody want to manipulate the PosixShmMemoryProvider in our codebase. I see only read-access to it where the const function is good for..
@elfenpiff what do you think? Do we need that maybe for future cases?

Reply:
@dkroenke @JimmyBch maybe, but we should follow the YAGNI principle (you aint gonna need it) and therefore we introduce this method when we require it. Till then we should remove it.

Reply:
Therefore, should I remove non-const method and revert change in const method ?

Reply:
@JimmyBch Yes, it should be easy to remove it, if you run into trouble let us know.